### PR TITLE
add Pagination, Sorting, and Swagger to the existing controllers

### PR DESCRIPTION
**Pull request by Leanchers**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -27,34 +32,42 @@ public class BookController {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
+    @Hidden
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]: why it hidden?

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java (hunk 1 above)
Comment (Leanchers) [reply src/main/java/mate/academy/onlinebookstore/controller/BookController.java]: I thought it is documentation for customers

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -32,13 +37,15 @@ public BookDto save(@RequestBody @Valid CreateBookRequestDto requestDto) {
...
+    @Operation(summary = "Get book by id", description = "Get a book by id")
     public BookDto findById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> findAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]:
```suggestion
    public List<BookDto> findAll(@ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 3]
```diff
@@ -54,7 +61,10 @@ public void deleteById(@PathVariable Long id) {
     }
 
     @GetMapping("/search")
-    public List<BookDto> search(@Valid BookSearchParametersDto searchParameters) {
-        return bookService.search(searchParameters);
+    @Operation(summary = "Get all searched books",
+            description = "Get a list of all searched books.")
+    public List<BookDto> search(@Valid BookSearchParametersDto searchParameters,
+            Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]:
```suggestion
            @ParameterObject @PageableDefault Pageable pageable) {
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**